pkg/proxy/volume.go

```diff
@@ -247,3 +249,85 @@ func (p *Proxy) spdkVolumeUnmapMarkSnapChainRemovedSet(ctx context.Context, req
 	/* Not implemented */
 	return &emptypb.Empty{}, nil
 }
+
+func (p *Proxy) VolumeMaximumSnapshotCountSet(ctx context.Context, req *rpc.EngineVolumeMaximumSnapshotCountSetRequest) (resp *emptypb.Empty, err error) {
+	log := logrus.WithFields(logrus.Fields{
+		"serviceURL":         req.ProxyEngineRequest.Address,
+		"engineName":         req.ProxyEngineRequest.EngineName,
+		"volumeName":         req.ProxyEngineRequest.VolumeName,
+		"backendStoreDriver": req.ProxyEngineRequest.BackendStoreDriver,
+	})
+	log.Infof("Setting volume flag MaximumSnapshotCount to %v", req.Count.Count)
+
+	switch req.ProxyEngineRequest.BackendStoreDriver {
+	case rpc.BackendStoreDriver_v1:
+		return p.volumeMaximumSnapshotCountSet(ctx, req)
+	case rpc.BackendStoreDriver_v2:
+		return p.spdkVolumeMaximumSnapshotCountSet(ctx, req)
+	default:
+		return nil, grpcstatus.Errorf(grpccodes.InvalidArgument, "unknown backend store driver %v", req.ProxyEngineRequest.BackendStoreDriver)
+	}
+
+}
+
+func (p *Proxy) volumeMaximumSnapshotCountSet(ctx context.Context, req *rpc.EngineVolumeMaximumSnapshotCountSetRequest) (resp *emptypb.Empty, err error) {
+	c, err := eclient.NewControllerClient(req.ProxyEngineRequest.Address, req.ProxyEngineRequest.VolumeName,
+		req.ProxyEngineRequest.EngineName)
+	if err != nil {
+		return nil, err
+	}
+	defer c.Close()
+
+	err = c.VolumeMaximumSnapshotCountSet(int(req.Count.Count))
+	if err != nil {
+		return nil, err
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (p *Proxy) spdkVolumeMaximumSnapshotCountSet(ctx context.Context, req *rpc.EngineVolumeMaximumSnapshotCountSetRequest) (resp *emptypb.Empty, err error) {
+	/* Not implemented */
+	return &emptypb.Empty{}, nil
+}
```

Comment:
We don't need space management for v2 volume, since it is using the whole disk instead. That means it should have itself space management inside.